packages/transit-vehicle-overlay/src/utils/state.js

```diff
@@ -0,0 +1,160 @@
+import { useState, useRef, useEffect, useCallback } from "react";
+import { checkRefreshInteval } from "./fetch";
+import * as data from "./data";
+
+/**
+ * use this hook when you want your layer to re-paint after otp-ui map zoom events fire
+ * this component will capture and save zoom state from otp-ui's base-map component to state
+ * :return: zoom level (number from leaflet) and the callback for base-map's onViewportChanged
+ */
+export function zoomState(initialZoom = 14) {
```

Comment:
For this (and any other hooks you define), I think it would be good for legibility to use the standard `useXyz` naming convention: https://reactjs.org/docs/hooks-custom.html#extracting-a-custom-hook